### PR TITLE
feat(dns): hierarchical compose DNS names matching OrbStack scheme

### PR DESCRIPTION
**Pull request by AprilNEA**
## Summary

- Compose containers now register a hierarchical service-level DNS alias (`<service>.<project>.arcbox.local`) in addition to the flat container-level alias (`<container-name>.arcbox.local`), matching OrbStack's `.orb.local` DNS scheme.
- Compose metadata is extracted from Docker labels (`com.docker.compose.project` / `com.docker.compose.service`) instead of parsing container names.
- Host-side `dns_entries` changed from `HashMap<String, String>` to `HashMap<String, Vec<String>>` to support multiple aliases per container.
- Both guest-side (in-VM DNS server) and host-side (daemon DNS forwarder) registration paths updated.

### DNS naming after this change

| Scenario | DNS name |
|----------|----------|
| Compose service-level | `web.myproject.arcbox.local` |
| Compose container-level | `myproject-web-1.arcbox.local` |
| Plain container | `mycontainer.arcbox.local` |

## Test plan

- [x] Unit tests for `collect_aliases` (compose v1/v2, plain, edge cases)
- [x] Unit tests for `extract_compose_info` (present/missing/empty labels)
- [x] Unit tests for `extract_container_dns_info` (compose, plain, fallback, no-ip, no-name)
- [x] Integration tests for alias extraction
- [ ] Manual: `docker compose up` a multi-service project → verify `service.project.arcbox.local` resolves
- [ ] Manual: plain `docker run --name foo` → verify `foo.arcbox.local` resolves

### CODE REVIEW

File: app/arcbox-core/src/runtime.rs [hunk 1]
```diff
@@ -784,34 +784,36 @@ impl Runtime {
...
-        tracing::info!(container_id, hostname, "DNS entry deregistered");
-        Some(hostname)
+    /// Removes all DNS entries for a container by its canonical ID.
+    pub async fn deregister_dns_by_id(&self, container_id: &str) {
+        let Some(hostnames) = self.dns_entries.write().await.remove(container_id) else {
+            return;
+        };
+        for hostname in &hostnames {
+            self.network_manager.deregister_dns(hostname);
+        }
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply app/arcbox-core/src/runtime.rs]: Fixed in 0805aff. Applied a similar approach — deregister_dns_by_id now holds the write lock while scanning remaining entries, and only deregisters hostnames no longer referenced by any container.

File: app/arcbox-core/src/runtime.rs [hunk 2]
```diff
@@ -784,34 +784,36 @@ impl Runtime {
...
-        self.network_manager.deregister_dns(&hostname);
-        tracing::info!(container_id, hostname, "DNS entry deregistered");
-        Some(hostname)
+    /// Removes all DNS entries for a container by its canonical ID.
+    pub async fn deregister_dns_by_id(&self, container_id: &str) {
+        let Some(hostnames) = self.dns_entries.write().await.remove(container_id) else {
+            return;
+        };
+        for hostname in &hostnames {
+            self.network_manager.deregister_dns(hostname);
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply app/arcbox-core/src/runtime.rs]: Fixed in 0805aff. Host-side deregister_dns_by_id now scans remaining dns_entries and only removes hostnames from the network manager when no other container still references them.

File: guest/arcbox-agent/src/docker_events.rs [hunk 3]
```diff
@@ -140,31 +140,26 @@ async fn handle_event(dns: &GuestDnsServer, json_str: &str) -> anyhow::Result<()
...
-                dns.deregister_container(name).await;
-                // Also deregister compose service alias.
-                for alias in crate::dns::collect_aliases(name) {
-                    if alias != name {
-                        dns.deregister_container(&alias).await;
-                    }
+                let compose = attrs.as_object().and_then(crate::dns::extract_compose_info);
+                for alias in crate::dns::collect_aliases(name, compose.as_ref()) {
+                    dns.deregister_container(&alias).await;
                 }
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply guest/arcbox-agent/src/docker_events.rs:151]: Fixed in 0805aff. Guest-side GuestDnsServer now tracks alias ownership per container via `alias_owners: HashMap<alias, HashMap<owner, IP>>`. Deregistering removes only the departing container's ownership; the alias persists (re-pointed to a remaining replica's IP) until all owners are gone.
